### ISSUE TITLE
PnP Modern Search Webpart | Help required in following WCAG Guidelines

### ISSUE THREAD

**Issue by ClientDev-ShubhrJha** [microsoft-search/pnp-modern-search#3439]
**Version used**
Ex: Version: 4.8.0
Web Part instance ID: 06df0296-c132-4072-8516-124ad9051077

**Describe the bug**
We are using PnP Modern Search and its related webpart. In these webpart we're facing issues with accessbility i.e. WCAG (Web Content Accessibility Guidelines).

PnP Filter Webpart
1. Missing alt text for linked images.
2. Interactive elements doesn't gain keyboard focus and are not accessible via keyboard.
3. Filters having checkboxes are not grouped using fieldset & legend.
4. Filters having datepicker component cannot be accessed, activated or controlled using keyboard alone - the field simply acts as a text input.

PnP Results Webpart
1. Current pagination doesn't have attributes as a result it is not communicated to assistive technology.

**To Reproduce**
**PnP Filter Webpart**
1. Configure a filter webpart which has checkboxes (multiple selection are made) with AND/OR filters.
2. Configure a filter webpart which has date picker.

3. Select multiple option in filter/checkboxes, this will add selected values appear near AND/OR options ->
- Now to cancel off filters user need to click on on 'x' button in the displayed selected options as a pill. This 'x' button cannot be accessed via Keyboard.
- 'x' button doesn't have alt text.
- Displayed option/pill doesn't gain keyboard focus.
- Upon using TAB key from keyboard filter/metadata heading can only be accessed via TAB, but for checkboxes need to use arrow it should work with TAB key as it is an interactive element.
- Within 1 metadata all available options/checkboxes should be grouped using fieldset and legends.
- AND/OR options are are grouped.
- Filter search combox boxes are not labelled meaning they will be read as "input, blank".
- When a filter is selected focus is returned to the top of the filter set and no feedback is provided meaning a user might believe their action has failed and not understand why focus has moved to where it is.

4. Go to Date picker filter/metadata
- The datepicker component cannot be accessed, activated or controlled using keyboard alone - the field simply acts as a text input.
- Date input fields are not labelled meaning they will be read as "input, blank".

**PnP Search Result Webpart**
1. The current page in the pagination should have aria-current="true" so that it's state is communicated to assistive technology.
2. Results in result webpart contains sharepoint icons  as they don't add any additional context to the page and are repetitive so should be treated as decorative and be given a blank alt attribute so they are not read by assistive technology.

**PnP Search Box Webpart**
1. Focus is not visible on the search submit button while navigating using TAB key on keyboard.

**Expected behavior**
Control or elements should align with WCAG guidelines.

**Screenshots**
NA

**Desktop (please complete the following information):**
 - Browser: Chrome/Edge/Chromium

**Additional context**
If a meeting is schedule this will help us to further resolve these issues faster as this is affecting Go Live date. Thank you.
